app/screens/home/recent_mentions/recent_mentions.tsx

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {useIsFocused, useRoute} from '@react-navigation/native';
+import React, {useCallback, useState, useEffect, useMemo} from 'react';
+import {useIntl} from 'react-intl';
+import {StyleSheet, View, ActivityIndicator, FlatList} from 'react-native';
+import Animated, {useAnimatedStyle, useSharedValue, withTiming} from 'react-native-reanimated';
+import {SafeAreaView, Edge} from 'react-native-safe-area-context';
+
+import {getRecentMentions} from '@actions/remote/search';
+import DateSeparator from '@app/components/post_list/date_separator';
+import {UserModel} from '@app/database/models/server';
+import {getDateForDateLine, isDateLine, selectOrderedPosts} from '@app/utils/post_list';
+import NavigationHeader from '@components/navigation_header';
+import {useServerUrl} from '@context/server';
+import {useTheme} from '@context/theme';
+import {useCollapsibleHeader} from '@hooks/header';
+
+import EmptyState from './components/empty';
+import Mention from './components/mention';
+
+import type PostModel from '@typings/database/models/servers/post';
+
+const AnimatedFlatList = Animated.createAnimatedComponent(FlatList);
+
+const EDGES: Edge[] = ['bottom', 'left', 'right'];
+
+type Props = {
+    currentTimezone: string | null;
+    currentUser: UserModel;
+    isTimezoneEnabled: boolean;
+    mentions: PostModel[];
+}
+
+const styles = StyleSheet.create({
+    flex: {
+        flex: 1,
+    },
+    empty: {
+        alignItems: 'center',
+        flex: 1,
+        justifyContent: 'center',
+    },
+});
+
+const RecentMentionsScreen = ({mentions, currentUser, currentTimezone, isTimezoneEnabled}: Props) => {
+    const theme = useTheme();
+    const route = useRoute();
+    const isFocused = useIsFocused();
+    const {formatMessage} = useIntl();
+    const [refreshing, setRefreshing] = useState(false);
+    const [loading, setLoading] = useState(true);
+    const serverUrl = useServerUrl();
+
+    const params = route.params! as {direction: string};
+    const toLeft = params.direction === 'left';
+    const translateSide = toLeft ? -25 : 25;
+    const opacity = useSharedValue(isFocused ? 1 : 0);
+    const translateX = useSharedValue(isFocused ? 0 : translateSide);
+
+    const title = formatMessage({id: 'screen.mentions.title', defaultMessage: 'Recent Mentions'});
+    const subtitle = formatMessage({id: 'screen.mentions.subtitle', defaultMessage: 'Messages you\'ve been mentioned in'});
+    const isLargeTitle = true;
+
+    useEffect(() => {
+        setLoading(true);
+        getRecentMentions(serverUrl).finally(() => {
+            setLoading(false);
+        });
```

Comment:
you could create a function with this and then call that function here and also in `handleRefresh`

Reply:
yes but loading is different than refreshing, isn't it?

Reply:
ahhh yes, I see.. you are using a different state